source/documentation/how-to/ha-vms.html.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Highly Available VMs
+authors: mzamazal
+category: howto
+layout: toc
+---
+
+# Highly Available VMs
+
+## What is a highly available VM?
+
+Highly available VMs (*HA* VMs) are VMs restarted by oVirt automatically whenever they disappear accidentally, such as due to failure of the host running the VM, storage problems or QEMU crash.
+
+## How to make a VM highly available
+
+High availability settings are present in the VM editing dialog.  Make sure to show Advanced Options there and select *High Availability*.
+
+![](/images/wiki/ha-vm-edit.png)
+
+You can see there are several settings there.  The primary one is *Highly Available* check box that must be checked.  Some of the additional settings are discussed later in this document.
+
+## Are highly available VMs always restarted automatically?
+
+Starting a VM automatically is a bit more complicated than one might initially think.  Several conditions must be satisfied in order to start such a VM safely and consistently with user actions.
+
+The very first condition is that oVirt Engine must be fully certain that the VM is not running anywhere at the moment.  Failing to satisfy this condition could lead to multiple instances of the VM running concurrently on different hosts, so called *split brain* situation.  oVirt uses safety mechanisms such as sanlock data, VM leases and resume behaviors to prevent split brain.  But it's still important to be aware that if oVirt Engine doesn't know (under certain setups) for sure that a VM is not already running, it can't start it automatically elsewhere.  For instance, if a host running a VM is unreachable it doesn't mean that the VM is not happily running there.  If corresponding protection mechanisms are not enabled then oVirt Engine may not be able to start the VM elsewhere, despite it is highly available.
+
+VMs manually shut down or powered off from oVirt or from the guest operating system are not restarted automatically.  Switching VMs off manually is a user action and oVirt honors it.  The user is responsible for starting such a VM again manually.
+
+There may be other reasons preventing start of a highly available VM, such as impossibility to run the VM due to resource constraints at a given moment.
```

Comment:
might be a good place to add a link to the recent devconf presentation from Arik here about this topic: https://www.youtube.com/watch?v=Gh2eB06IE2A